frontend/packages/network-attachment-definition-plugin/src/types/index.ts

```diff
@@ -0,0 +1,26 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+export type IPAMConfig = {
+  type?: string;
+  subnet?: string;
+  dataDir?: string;
+};
+
+export type NetworkAttachmentDefinitionConfig = {
+  cniVersion?: string;
+  name?: string;
+  type?: string;
+  bridge?: string;
+  isGateway?: true;
+  ipam?: IPAMConfig;
+  plugins?: NetworkAttachmentDefinitionConfig[];
+};
+
+// The config is a JSON object with the NetworkAttachmentDefinitionConfig type above stored as a string
+export type NetworkAttachmentDefinitionSpec = {
+  config: string;
+};
+
+export type NetworkAttachmentDefinitionKind = {
+  spec?: NetworkAttachmentDefinitionSpec;
```

Comment:
is `spec` optional?

Reply:
From what I read in the network standard CRD doc it is. I may have misread it, though. I'll double-check.